cmd/rpcdaemon/commands/eth_receipts.go

```diff
@@ -646,6 +647,15 @@ func (api *APIImpl) GetTransactionReceipt(ctx context.Context, txnHash common.Ha
 		return nil, fmt.Errorf("getReceipts error: %w", err)
 	}
 
+	var edg *big.Int
+	if n := block.Number().Uint64(); n > 0 {
+		if parentHeader, err := api._blockReader.HeaderByNumber(ctx, tx, n-1); err != nil {
```

Comment:
please use `Header()` instead of `HeaderByNumber` because you know parentHash here